mantidproject/mantid
# Provide better sorting for CropToComponent

Fixes #18423 

With the initial issue description we wanted to achieve a better ordering of the cropped workspace in order to be inline with the original workspace. Initially we thought this would be achieved via a new ordering option. It turns out that this was not the best idea, in fact the old `OrderByDetId` option is not very useful either. Instread we map the crop detIds to the workspace indices of the workspace and achieve the correct order in this way. This makes the algorithm much simpler.

### To test:

Run the doc test. You can play with it and ensure that it makes sense.

```python
# Create sample workspace with four banks where each bank has 3x3 detectors
sample_workspace = CreateSampleWorkspace(NumBanks=4, BankPixelWidth=3)

# Crop to a component, we select bank2 here
cropped_workspace = CropToComponent(InputWorkspace=sample_workspace, ComponentNames="bank2")

# Check the number of histograms
sample_number_of_histograms = sample_workspace.getNumberHistograms()
cropped_number_of_histograms = cropped_workspace.getNumberHistograms()

print("The original workspace has {0} histograms and the cropped workspace has {1} histograms.".format(sample_number_of_histograms, cropped_number_of_histograms))
```

### Release notes

See [here](https://github.com/mantidproject/mantid/commit/e2d9bba4a85c7e17f09ff3ca894bce99b7b8b20c)
---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
